server/test/unit/server/webservices/views/test_consumers.py

```diff
@@ -0,0 +1,1377 @@
+                                    '_href': '/v2/consumers/foo/bindings/some-repo/'}
+
+        request = mock.MagicMock()
+        consumer_bindings = ConsumerBindingsView()
+        response = consumer_bindings.get(request, 'foo', 'some-repo')
+
+        expected_cont = [{'consumer_id': 'foo', '_href': '/v2/consumers/foo/bindings/some-repo/',
+                          'repo_id': 'some-repo'}]
+
+        mock_resp.assert_called_once_with(expected_cont)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.consumers.factory')
+    def test_get_consumer_bindings_invalid_consumer(self, mock_factory):
+        """
+        Test all bindings retrieval invalid consumer
+        """
+        mock_factory.consumer_manager.return_value.get_consumer.side_effect = MissingResource()
+
+        request = mock.MagicMock()
+        consumer_bindings = ConsumerBindingsView()
+
+        try:
+            response = consumer_bindings.get(request, 'nonexistent_id')
+        except MissingResource, response:
+            pass
+        else:
+            raise AssertionError("MissingResource should be raised with nonexistent consumer_id")
+        self.assertEqual(response.http_status_code, 404)
+        self.assertEqual(response.error_data['resources'], {'consumer_id': 'nonexistent_id'})
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.consumers.factory')
+    def test_get_consumer_bindings_invalid_repo(self, mock_factory):
+        """
+        Test all bindings retrieval invalid repo
+        """
+        mock_factory.repo_query_manager.return_value.find_by_id.return_value = None
+
+        request = mock.MagicMock()
+        consumer_bindings = ConsumerBindingsView()
+
+        try:
+            response = consumer_bindings.get(request, 'foo', 'some-repo')
+        except MissingResource, response:
+            pass
+        else:
+            raise AssertionError("MissingResource should be raised with nonexistent repo_id")
+        self.assertEqual(response.http_status_code, 404)
+        self.assertEqual(response.error_data['resources'], {'repo_id': 'some-repo'})
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    @mock.patch('pulp.server.webservices.views.consumers.consumer_task.bind')
+    def test_create_binding_async(self, mock_bind):
+        """
+        Test bind consumer to a repo async task.
+        """
+        request = mock.MagicMock()
+        request.body = json.dumps({'repo_id': 'xxx', 'distributor_id': 'yyy'})
+        consumer_bindings = ConsumerBindingsView()
+        self.assertRaises(OperationPostponed, consumer_bindings.post, request, 'test-consumer')
```

Comment:
Good idea. It is pretty well established what OperationPostponed does, I don't think we need the extra assertions about the response. This is nice.